Paintroid/src/main/java/org/catrobat/paintroid/dialog/MainActivityDialogFragment.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Paintroid: An image manipulation application for Android.
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.dialog
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatDialogFragment
+import org.catrobat.paintroid.contract.MainActivityContracts
+import org.catrobat.paintroid.contract.MainActivityContracts.MainView
+
+open class MainActivityDialogFragment : AppCompatDialogFragment() {
+    lateinit var presenter: MainActivityContracts.Presenter
+        private set
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        val activity = activity as MainView?
+            ?: throw IllegalArgumentException("Parent activity must implement MainActivityContracts.MainView!")
```

Comment:
What do you think about creating a string resource for this message? Or maybe a top-level constant?

Reply:
Good idea!